### ISSUE TITLE
Issues with experimental loadQuery in latest release

### ISSUE THREAD

**Issue by santino** [facebook/relay#3132]
Latest Relay experimental release `0.0.0-experimental-7f19cb58` marked `preloadQuery` as deprecated.  
Alternative is to use `loadQuery` but there are a couple of issues with that.

1. `loadQuery` export is broken
2. `loadQuery` forces to use persisted queries

As about point 1 `react-relay/hooks` exports `loadQuery` like so:
```js
// react-relay/hooks.js
...
const loadQuery = require('./relay-experimental/loadQuery');
...

module.exports = {
  ...
  loadQuery: loadQuery,
  ...
};
```

This would assume `loadQuery` is default export in the relevant module of `relay-experimental` package; but this is not the case, given the module exports two functions `module.exports = {loadQuery, useTrackLoadQueryInRender};`.  
As such importing as per documentation is not going to work `const { loadQuery } = require('react-relay/hooks')`.

My current workaround to use `loadQuery` is
```js
import { loadQuery as defaultLoadQuery } from 'react-relay/hooks'
...
const { loadQuery } = defaultLoadQuery
```

As about point 2 I'm totally sold on usage of persisted queries in production and when you have full control of your GraphQL server.  
However this is not always the case, if you're using a public GraphQL API you would not be able to use your generated `persistedQueries.json` on the server.
Even when you are working with your own GraphQL server during development phase you will most likely play with your queries several times; using persisted queries in dev environment forces you to have a sync mechanisms for the `persisteQueries.json` between client and server. This is not necessarily what devs need, especially on an environment which main concern is debugging and not performance, hence the benefits of persisted queries are probably not relevant.

```js
// relay-experimental/loadQuery.js

  ({id: moduleId} = params);
    invariant(
      moduleId !== null,
      'Relay: `loadQuery` requires that preloadable query `%s` has a persisted query id',
      params.name,
    );
```
The above seems like a debatable limitation on something that would otherwise work just fine when commenting out.

EDIT: upon further investigation I see #3125 addressed this a couple of days ago; would be great to have this change released.

**Comment by josephsavona** [comment]
Thanks for posting. Persisted queries are not required for `loadQuery` to work, and as you noted #3125 addressed this. We're currently a bit short on staff due to vacations etc, and will publish a new release as time permits.

**Comment by santino** [comment]
Thanks @josephsavona 
Limited resourcing is totally understandable and I was happy to find an amend for point 2.

How about point 1?
Any plan to fix the loadQuery export?

**Comment by sibelius** [comment]
can you send a pr to fix point 1?

**Comment by santino** [comment]
I surely can.
It depends how you want to fix it.

I'm not sure how you use `loadQuery` and `useTrackLoadQueryInRender` internally; personally right now I would need only the first one.

Some possible simple solutions I can think of:
1. export only `loadQuery` as default export from `relay-experimental/loadQuery.js`
2. keep existing export `{loadQuery, useTrackLoadQueryInRender}` from `relay-experimental/loadQuery.js` and fix export in `react-relay/hooks` into `loadQuery: loadQuery.loadQuery,`
3. export both functions from `react-relay/hooks`: `loadQuery: loadQuery.loadQuery, useTrackLoadQueryInRender: loadQuery. useTrackLoadQueryInRender`

Happy to go with any of the above; let me know what makes more sense to you :)

**Comment by boo1-boo1** [comment]
This seems to be fixed in the code, but not yet published to npm.

**Comment by erictaylor** [comment]
See: https://github.com/facebook/relay/pull/3191

**Comment by benmurden** [comment]
I, too, am encountering this error, even though the [release notes suggest this made it to `v10.0.1`](https://github.com/facebook/relay/releases/tag/v10.0.1).

**Comment by fstream9** [comment]
I got this error too in v10.0.01 but is fixed when I changed to the experimental branch.

**Comment by sorenhoyer** [comment]
Hopefully this will be fixed in the next experimental release.

Currently using the workaround santino proposed.

```
import { loadQuery as defaultLoadQuery } from 'react-relay/hooks'
...
const { loadQuery } = defaultLoadQuery
```

**Comment by santino** [comment]
I updated react-relay today and confirm the issue is fixed.

Not sure which version introduced the fix, but currently using `"react-relay": "0.0.0-experimental-0b967d2e",` and things are working properly now; just for reference.

I can import, as expected, with `import { loadQuery } from 'react-relay/hooks'`
